packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts

```diff
@@ -60,24 +60,35 @@ export class CloudinaryMediaStore implements MediaStore {
     )
   }
   async list(options: MediaListOptions): Promise<MediaList> {
-    let query = '?'
-
-    if (options.directory) {
-      query += `directory=${encodeURIComponent(options.directory)}`
-    }
+    const query = this.buildQuery(options)
     const response = await this.fetchFunction('/api/cloudinary/media' + query)
-
-    const { items } = await response.json()
+    const { items, offset } = await response.json()
     return {
       items: items.map((item) => item),
-      totalCount: items.length,
-      limit: 500,
-      offset: 0,
-      nextOffset: undefined,
+      nextOffset: offset,
     }
   }
 
   previewSrc = (publicId: string) => publicId
 
   parse = (img) => img.previewSrc
+
+  private buildQuery(options: MediaListOptions) {
+    const entries = Object.entries(options)
+
+    entries.forEach((entry) => {
+      const key = entry[0]
+      const value = entry[1]
+
+      if (value === '' || value === undefined) {
+        delete options[key]
+      }
+    })
+
+    const params = Object.keys(options)
```

Comment:
you could do something like the below and remove that whole `forEach` loop:

```suggestion
    const params = Object.keys(options).filter((key) => (options[key] !== '' && options[key] !== undefined))
```
...and then chain that into the `map` function.

This has the added benefit of avoiding the mutation on `options`. I think the `delete` keyword can actually mutate option arguments from the caller end! So we probably want to avoid that.

Reply:
I agree, that's a much more straight forward approach.